src/ray/core_worker/store_provider/local_plasma_provider.h

```diff
@@ -0,0 +1,71 @@
+#ifndef RAY_CORE_WORKER_LOCAL_PLASMA_STORE_PROVIDER_H
+#define RAY_CORE_WORKER_LOCAL_PLASMA_STORE_PROVIDER_H
+
+#include "plasma/client.h"
+#include "ray/common/buffer.h"
+#include "ray/common/id.h"
+#include "ray/common/status.h"
+#include "ray/core_worker/common.h"
+#include "ray/core_worker/store_provider/store_provider.h"
+#include "ray/raylet/raylet_client.h"
+
+namespace ray {
+
+class CoreWorker;
+
+/// The class provides implementations for accessing plasma store, which includes both
+/// local and remote store, remote access is done via raylet.
+class CoreWorkerLocalPlasmaStoreProvider : public CoreWorkerStoreProvider {
+ public:
+  CoreWorkerLocalPlasmaStoreProvider(const std::string &store_socket);
+
+  /// Put an object with specified ID into object store.
+  ///
+  /// \param[in] object The ray object.
+  /// \param[in] object_id Object ID specified by user.
+  /// \return Status.
+  Status Put(const RayObject &object, const ObjectID &object_id) override;
+
+  /// Get a list of objects from the object store.
+  ///
+  /// \param[in] ids IDs of the objects to get.
+  /// \param[in] timeout_ms Timeout in milliseconds, wait infinitely if it's negative.
+  /// \param[in] task_id ID for the current task.
+  /// \param[out] results Result list of objects data.
+  /// \return Status.
+  Status Get(const std::vector<ObjectID> &ids, int64_t timeout_ms, const TaskID &task_id,
+             std::vector<std::shared_ptr<RayObject>> *results) override;
+
+  /// Wait for a list of objects to appear in the object store.
+  ///
+  /// \param[in] IDs of the objects to wait for.
+  /// \param[in] num_returns Number of objects that should appear.
+  /// \param[in] timeout_ms Timeout in milliseconds, wait infinitely if it's negative.
+  /// \param[in] task_id ID for the current task.
+  /// \param[out] results A bitset that indicates each object has appeared or not.
+  /// \return Status.
+  Status Wait(const std::vector<ObjectID> &object_ids, int num_objects,
+              int64_t timeout_ms, const TaskID &task_id,
+              std::vector<bool> *results) override;
+
+  /// Delete a list of objects from the object store.
+  ///
+  /// \param[in] object_ids IDs of the objects to delete.
+  /// \param[in] local_only Whether only delete the objects in local node, or all nodes in
+  /// the cluster.
+  /// \param[in] delete_creating_tasks Whether also delete the tasks that
+  /// created these objects. \return Status.
```

Comment:
Not well formatted.

Reply:
Instead of copying the comments from super class, I think it's okay to say ``See `CoreWorkerStoreProvider::Delete` for semantics`` here. But more importantly, we should document clearly the differences with the super class. For example, "local_only must be true, and delete_creating_tasks must be false here"

Reply:
I think this is a good point. 
Actually I'm considering to refactor it a bit so that the `ObjectInterface` takes care of cross-node scenario, and have `plasma provider` to focus only on local store, this should make the interfaces cleaner - but I'm feeling the feasibility is more clear after the `in-memory store provider `is fully implemented, so I'm planning to delay this refactor then.

Reply:
yeah, that refactor sounds reasonable.